### DISCUSSION TITLE
GPU pass-through for Jellyfin

### DISCUSSION THREAD

**Discussion by neilkpd** [Jip-Hop/jailmaker#60] [Q&A]
System:
Chassis: Supermicro SC846BE1C-R1K23B
MB: Supermicro X11SPH-nCTF
CPU: Intel® Xeon® Silver 4114 Processor
RAM: 256 GiB Samsung M393A4K40BB2-CTD
GPU: Asus Geforce RTX 3050
OS: TrueNAS-SCALE-23.10.0.1
System Settings > Advanced > Isolated GPU Device(s) - Isolated GPU Device(s):NVIDIA Corporation GA106 [Geforce RTX 3050]
Jailmaker (jlmkr.py 1.0.1)
Docker - installed per convenience script

Jellyfin docker-compose.yml
version: '3.9'
services:
  jellyfin:
    image: jellyfin/jellyfin:latest
    container_name: jellyfin
    networks:
      - proxy
    security_opt:
      - no-new-privileges:true
    restart: unless-stopped
    user: 3000:3000
    environment:
      - PUID=3000
      - PGID=3000
      - TZ=America/New_York
    volumes:
      - /mnt/tank1/apps/jellyfin/config:/config
      - /mnt/tank1/apps/jellyfin/cache:/cache
      - /mnt/tank1/apps/jellyfin/logs:/logs
      - /mnt/tank0/media:/data/media
      #- /dev/shm:/data/transcode # Offload transcoding to RAM if you have enough RAM
    devices:
    - /dev/dri/card0:/dev/dri/card0
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.jellyfin.entrypoints=https"
      - "traefik.http.routers.jellyfin.rule=Host(`jellyfin.example.net`)"
      - "traefik.http.routers.jellyfin.tls=true"
      - "traefik.http.routers.jellyfin.service=jellyfin"
      ## HTTP Services
      - "traefik.http.services.jellyfin.loadbalancer.server.port=8096"
networks:
  proxy:
    external: true

Jellyfin > Playback > Transcoding > Hardware acceleration > Nvidia NVENC
(H264, HEVC, VC1, AV1, HEVC 10bit)
I get Playback Error:
This client isn't compatible with the media and server isn't sending a compatible media format.

And without Hardware acceleration, the same media is played.
I would appreciate some help here!

**Comment by Jip-Hop** [comment]
What is the output of nvidia-smi on the TrueNAS host? What is the output or nvidia-smi in the jail? Also post your jail config file please.

**Reply by Jip-Hop** [reply]
Or are you trying to use the Intel GPU instead of nvidia?

**Reply by neilkpd** [reply]
nvidia-smi:
TrueNAS host
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

jail
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

jail config:
startup=1
docker_compatible=1
gpu_passthrough_intel=0
gpu_passthrough_nvidia=0
systemd_nspawn_user_args=

You generally will not need to change the options below
systemd_run_default_args=--property=KillMode=mixed --property=Type=notify --property=RestartForceExitStatus=133 --property=SuccessExitStatus=133 --property=Delegate=yes --property=TasksMax=infinity --collect --setenv=SYSTEMD_NSPAWN_LOCK=0
systemd_nspawn_default_args=--keep-unit --quiet --boot


Hardware acceleration: 
Nvidia NVENC

**Reply by Jip-Hop** [reply]
If nvidia-smi doesn't work on the host then it will not work inside the jail either. I suspect your nvidia GPU isn't supported by the nvidia driver included with SCALE and you're going to have to wait for a software update. When nvidia-smi does work on the host and you want to pass it through to the jail, you have to enable GPU passthrough during jail setup. You now have gpu_passthrough_nvidia=0 so your GPU isn't available inside the jail. Also I think you need to add the nvidia GPU to devices in the compose file. Now you only added /dev/dri/card0.

By the way what does this do?
System Settings > Advanced > Isolated GPU Device(s) - Isolated GPU Device(s):NVIDIA Corporation GA106 [Geforce RTX 3050]

Perhaps nvidia-smi doesn't work because you have isolated the GPU?

**Comment by neilkpd** [comment]
The Geforce RTX 3050 was isolated when:
I installed OS: TrueNAS-SCALE-23.10.0
Jailmaker (jlmkr.py 1.0)
Debian Bookworm
During Jailmaker installation, I was given the option to pass-through the onboard Intel GPU and the Geforce RTX 3050.
But I did not proceed as the updated versions of both software were released.
